rskj-core/src/test/java/co/rsk/net/sync/SyncMessageHandlerTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2024 RSK Labs Ltd.
+ * (derived from ethereumJ library, Copyright (c) 2016 <ether.camp>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.net.sync;
+
+import co.rsk.net.Peer;
+import co.rsk.net.messages.Message;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+class SyncMessageHandlerTest {
+
+    private static final long THREAD_JOIN_TIMEOUT = 10_000; // 10 secs
+
+    private BlockingQueue<SyncMessageHandler.Job> jobQueue;
+    private Thread thread;
+    private volatile Boolean isRunning;
+
+    private final SyncMessageHandler.Listener listener = mock(SyncMessageHandler.Listener.class);
+
+    @BeforeEach
+    void setUp() {
+        jobQueue = new LinkedBlockingQueue<>();
+        thread = new Thread(new SyncMessageHandler("SNAP requests", jobQueue, listener) {
+
+            @Override
+            public boolean isRunning() {
+                return isRunning;
+            }
+        }, "snap sync request handler");
+    }
+
+    @Test
+    void run_processOneJob() throws InterruptedException {
+        //given
+        final AtomicBoolean jobCalled = new AtomicBoolean();
+
+        isRunning = Boolean.TRUE;
+        doAnswer(invocation -> {
+            isRunning = Boolean.FALSE;
+            return null;
+        }).when(listener).onQueueEmpty();
+
+        thread.start();
+
+        //when
+        jobQueue.put(new SyncMessageHandler.Job(mock(Peer.class), mock(Message.class)) {
+            @Override
+            public void run() {
+                jobCalled.set(true);
+            }
+        });
+
+        //then
+        thread.join(THREAD_JOIN_TIMEOUT);
```

Comment:
Mini idea: I would move this `thread.join(...)` to the `//when` block as it's not part of the "check" step.

Reply:
Same [here](https://github.com/rsksmart/rskj/pull/2500/files#diff-dc006ec78d028f9a5c118de85bf07d9eda8cefb44e46bb9273e795ef99573484R116) and [here](https://github.com/rsksmart/rskj/pull/2500/files#diff-dc006ec78d028f9a5c118de85bf07d9eda8cefb44e46bb9273e795ef99573484R138).